modules/mod-aup/ImportAUP.cpp

```diff
@@ -557,6 +557,21 @@ void AUPImportFileHandle::HandleXMLEndTag(const std::string_view& tag)
    {
       node.handler->HandleXMLEndTag(tag);
    }
+   if(tag == "wavetrack")
+   {
+      auto& tracks = TrackList::Get(mProject);
+      const auto first = *tracks.Find(mWaveTrack);
+      if(first != mWaveTrack)
+      {
+         if(auto err = first->GetErrorOpening())
+            SetError(*err);
+         if(!first->LinkConsistencyFix(true))
+            SetError(XO("A channel of a stereo track was missing."));
+
+         if(mHasParseError)
+            return;
+      }
+   }
```

Comment:
I'm convinced that this is correct to repair the track consistency, but there is also new logic in `ProjectFileManager::ReadProjectFile` to warn the user about the change, and that function is not reached.

It should not be difficult to make that reusable.

Reply:
My guess that the main reason to show this dialog is to warn about possible effect stack loss, when you import AUP with mismatched channels nothing gets removed

Reply:
But you are also changing it in #5260 to give other warnings about mismatched rates and formats

Reply:
The warning is about effect stacks that can be removed as we split channels, we don't change sample format or rate

Reply:
And existing texts do not fit in that case I think

Reply:
the warning contains two pieces of information: 
1. your project will look different to what you last saved because we migrated some stereo tracks to mono tracks
2. realtime effect stacks may be lost. 

In the case of legacy importers, part 2 is not relevant but part 1 is, so I'd still want to see the message box.